src/dune_rules/fdo.ml

```diff
@@ -168,21 +175,28 @@ module Linker_script = struct
     let linker_script_path =
       Path.Build.(relative ctx.build_dir (Path.to_string linker_script))
     in
+    let open Memo.Build.O in
     let flags () =
       let open Command.Args in
-      match get_profile ctx with
+      let* () = Memo.Build.return () in
```

Comment:
Just to make sure I understand, this is here to delay the computation of the following expression?

In any case, can't the `()` argument to flags be removed?

Reply:
Yes, binding on unit delays the synchronous part of the computation until the resulting `Memo.Build.t` is executed. I just realised that I actually had two such delays, so I deleted one of them.

The unit argument of `flags` is still useful: if we omit it, `get_profile` will always be executed, which I'm guessing is undesirable (I'm not the original author of this code but `Action_builder.delayed` must have been there for a reason).

Reply:
Good point, makes sense.